### PR TITLE
Strip png timestamps to ensure PNG output is consistent over time

### PR DESCRIPTION
**Pull request by byroot**
I red / greened it. The fingerprint was changing on every run before. It's now consistent.

It's likely still dependent on the imagemagick version though, so not sure if the local fingerprint will match Travis's. It might also break on random people computers if they try to run specs locally.

@rafaelfranca @wvanbergen for review please.

### CODE REVIEW

**Comment by wvanbergen** [comment]
Makes sense to me. Not sure why the test is failing though.

**Comment by byroot** [comment]
Yeah it's what I was afraid of. It fails because the version of imagemagick on travis is different from my local.

I think I can simply test that the output is consistent by compiling multiple times in a row with a `sleep` :disappointed:

**Comment by rafaelfranca** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by byroot** [comment]
Got a green!

I'll release a new version in a sec.

File: lib/sprockets/svg.rb [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
       image_list = Magick::Image.from_blob(svg_blob) { self.format = 'SVG' }
       image = image_list.first
       image.format = 'PNG'
-      image.to_blob
+      strip_png_metadata(image.to_blob)
+    end
+
+    def strip_png_metadata(png_blob)
+      stream = ChunkyPNG::Datastream.from_blob(png_blob)
+      image = ChunkyPNG::Image.from_datastream(stream)
```
Comment (wvanbergen) [inline lib/sprockets/svg.rb:22]: You can do `ChunkyPNG::Image.from_blob(blob)` to combine those two steps.

File: lib/sprockets/svg.rb [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
+      USELESS_PNG_METADATA.each(&image.metadata.method(:delete))
+      image.to_datastream.to_blob
```
Comment (wvanbergen) [inline lib/sprockets/svg.rb:24]: Same here, `image.to_blob` will work

Comment (byroot) [inline lib/sprockets/svg.rb:24]: Arf, I was looking for those, but missed them in your "excellent API docs" :trollface:

**Comment by wvanbergen** [comment]
Now, you can add `oily_png` to your gemspec and magically speed this up :trollface:

**Comment by byroot** [comment]
I'll let the users do it in their own apps, if they want to.

**Pull request merged**